File: src/main/java/io/github/takusan23/clickmanaita/Enchant/ClickManaitaEnchant.java
package io.github.takusan23.clickmanaita.Enchant;

import net.minecraft.enchantment.Enchantment;
import net.minecraft.enchantment.EnchantmentType;
import net.minecraft.inventory.EquipmentSlotType;
import net.minecraft.util.text.ITextComponent;

public class ClickManaitaEnchant extends Enchantment {

    protected ClickManaitaEnchant(Rarity rarityIn, EquipmentSlotType... slots) {
        super(rarityIn, EnchantmentType.BREAKABLE, slots);
    }

    public int getMinEnchantability(int enchantmentLevel) {
        return 1 + 10 * (enchantmentLevel - 1);
    }

    public int getMaxEnchantability(int enchantmentLevel) {
        return super.getMinEnchantability(enchantmentLevel) + 50;
    }


    public int getMaxLevel() {
        return 5;
    }

}

File: src/main/kotlin/io/github/takusan23/clickmanaita/enchant/ClickManaitaEnchant.kt
package io.github.takusan23.clickmanaita.enchant

import net.minecraft.enchantment.Enchantment
import net.minecraft.enchantment.EnchantmentTarget
import net.minecraft.entity.EquipmentSlot

/**
 * クリックまな板エンチャント。
 * */
class ClickManaitaEnchant : Enchantment(Rarity.COMMON, EnchantmentTarget.BREAKABLE, EquipmentSlot.values()) {

    companion object {

        /** クリックまな板エンチャント。エンチャント一個なのでエンチャント列挙用クラス作るのもあれなので */
        val CLICKMANAITA_ENCHANT = ClickManaitaEnchant()

    }

    override fun getMinPower(level: Int): Int = 1 + 10 * (level - 1)

    override fun getMaxPower(level: Int): Int = super.getMinPower(level) + 50

    override fun getMaxLevel(): Int = 5

}